environment-variables/src/main/java/com/palantir/gradle/utils/environmentvariables/EnvironmentVariables.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.utils.environmentvariables;
+
+import javax.inject.Inject;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
+
+public abstract class EnvironmentVariables {
```

Comment:
Should probably add a couple of basic tests for this, one which sets a property in a test invocation then checks it's got back and one that checks a real env var is used when `__TESTING` is not set.